geotrek/core/views.py

```diff
@@ -188,6 +192,66 @@ def dispatch(self, *args, **kwargs):
         return super(PathUpdate, self).dispatch(*args, **kwargs)
 
 
+class MultiplePathDelete(TemplateView):
+    template_name = "core/multiplepath_confirm_delete.html"
+    model = Path
+    success_url = "core:path_list"
+
+    def dispatch(self, *args, **kwargs):
+        self.paths_pk = self.kwargs['pk'].split(',')
+        self.paths = []
+        for pk in self.paths_pk:
+            path = Path.objects.get(pk=pk)
+            self.paths.append(path)
+            if path.draft and not self.request.user.has_perm('core.delete_draft_path'):
+                messages.warning(self.request, _(
+                    u'Access to the requested resource is restricted. You have been redirected.'))
+                return redirect('core:path_list')
+            if not path.draft and not self.request.user.has_perm('core.delete_path'):
+                messages.warning(self.request, _(
+                    u'Access to the requested resource is restricted. You have been redirected.'))
+                return redirect('core:path_list')
+            if not path.same_structure(self.request.user):
+                messages.warning(self.request, _(u'Access to the requested resource is restricted by structure. '
+                                                 u'You have been redirected.'))
+                return redirect('core:path_list')
+        return super(MultiplePathDelete, self).dispatch(*args, **kwargs)
+
+    # Add support for browsers which only accept GET and POST for now.
+    def post(self, request, *args, **kwargs):
+        return self.delete(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        for path in self.paths:
+            path.delete()
+        return HttpResponseRedirect(reverse(self.success_url))
+
+    def get_context_data(self, **kwargs):
+        context = super(MultiplePathDelete, self).get_context_data(**kwargs)
+        topologies_by_model = defaultdict(list)
+        for path in self.paths:
```

Comment:
Il faudrait factoriser le code à l'intérieur de ce for avec celui de la classe PathDelete

Reply:
Done